File: API/Helpers/UserParams.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Helpers
{
    public class UserParams : PaginationParams
    {

        public string CurrentUsername  { get; set; }
        public string Gender { get; set; }
        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 99;
        public string OrderBy { get; set; } = "lastActive";
        public bool ConcertFilter { get; set; }
        public int Distance { get; set; } = 10; // Distances range: 5, 10, 15, 25, 50, anywhere(100)
        public double Latitude { get; set; }
        public double Longitude { get; set; }
    }
}

File: client/src/app/_models/userParams.ts
import { User } from "./user";

export class UserParams {
    gender: string;
    minAge = 18;
    maxAge = 99;
    pageNumber = 0;
    pageSize = 6;
    orderBy = 'lastActive';
    concertFilter?: boolean;
    distance = 10;
    latitude = 0.0;
    longitude = 0.0;

    constructor(user: User) {
        this.gender = user.gender === 'female' ? 'male' : 'female';
        this.concertFilter = false;
    }
}